edkrepo/commands/list_pins_command.py

```diff
@@ -0,0 +1,123 @@
+            output_string = ''
+            separator = '\n'
+        cfg, user_cfg, conflicts = list_available_manifest_repos(config['cfg_file'], config['user_cfg_file'])
+        try:
+            manifest = get_workspace_manifest()
+            pull_workspace_manifest_repo(manifest, config['cfg_file'], config['user_cfg_file'], args.source_manifest_repo, False)
+            src_manifest_repo = find_source_manifest_repo(manifest, config['cfg_file'], config['user_cfg_file'], args.source_manifest_repo)
+            if src_manifest_repo in cfg:
+                manifest_directory = config['cfg_file'].manifest_repo_abs_path(src_manifest_repo)
+            elif src_manifest_repo in user_cfg:
+                manifest_directory = config['user_cfg_file'].manifest_repo_abs_path(src_manifest_repo)
+            else:
+                raise EdkrepoManifestNotFoundException(SOURCE_MANIFEST_REPO_NOT_FOUND.format(manifest.project_info.codename))
+        except EdkrepoWorkspaceInvalidException:
+            if not args.project:
+                raise EdkrepoInvalidParametersException(humble.NOT_IN_WKSPCE)
+            else:
+                #arg parse provides a list so only use the first item since we are limiting users to one project
+                manifest_repo, src_cfg, manifest_path = find_project_in_all_indices(args.project[0],
+                                                                                 config['cfg_file'],
+                                                                                 config['user_cfg_file'],
+                                                                                 PROJ_NOT_IN_REPO.format(args.project[0]),
+                                                                                 SOURCE_MANIFEST_REPO_NOT_FOUND.format(args.project[0]))
+                if manifest_repo in cfg:
+                    manifest_directory = config['cfg_file'].manifest_repo_abs_path(manifest_repo)
+                elif manifest_repo in user_cfg:
+                    manifest_directory = config['user_cfg_file'].manifest_repo_abs_path(manifest_repo)
+                manifest = ManifestXml(manifest_path)
+        if manifest.general_config.pin_path is None:
+            print(humble.NO_PIN_FOLDER)
+            return
+        pin_folder = os.path.normpath(os.path.join(manifest_directory, manifest.general_config.pin_path))
+        if args.verbose:
+            if not use_less:
+                print(humble.PIN_FOLDER.format(pin_folder))
+            else:
+                output_string = (humble.PIN_FOLDER.format(pin_folder))
+        for dirpath, _, filenames in os.walk(pin_folder):
+            for file in filenames:
+                pin_file = os.path.join(dirpath, file)
+                # Capture error output from manifest parser stdout to properly interleave with other pin data
+                stdout = sys.stdout
+                sys.stdout = io.StringIO()
+                pin = ManifestXml(pin_file)

```

Comment:
We should probably wrap this in a try/except block just in case the pin directory has files in it that are not pin files. If such a file is encountered, we should skip it and continue enumerating the remaining files.